### PR TITLE
[docs] Enable the Sphinx doc system for Cling

### PR DESCRIPTION
**Pull request by vgvassilev**
cc: @sarabellei

### CODE REVIEW

**Comment by vgvassilev** [comment]
We don't need a ci check for this as we do not have coverage.

**Comment by Axel-Naumann** [comment]
Should we have this in a dedicated directory, say `interpreter/cling/docs/sphinx` or `interpreter/cling/sphinx-docs`? `interpreter/cling/docs` currently contains doxygen files, release notes, and `CMakeLists.txt`.

File: interpreter/cling/docs/CMakeLists.txt [hunk 1]
```diff
@@ -57,8 +57,8 @@ endif()
 endif()
 
 if (LLVM_ENABLE_SPHINX)
+  include(AddSphinxTarget)
```
[the remark this answers is not in the document]
Comment (vgvassilev) [reply interpreter/cling/docs/CMakeLists.txt:60]: IIRC, that's the same as what was done for clang.

**Comment by vgvassilev** [comment]
> Should we have this in a dedicated directory, say `interpreter/cling/docs/sphinx` or `interpreter/cling/sphinx-docs`? `interpreter/cling/docs` currently contains doxygen files, release notes, and `CMakeLists.txt`.

It looks like `interpreter/llvm/src/tools/clang` have the same structure. I'd rather keep it like that as we are reusing documentation settings and build rules which then we will have to tweak and maintain ourselves.

**Comment by vgvassilev** [comment]
This landed in #11474

**Pull request closed without merging**